foreign-key.md

```diff
@@ -0,0 +1,312 @@
+  KEY `idx_pid` (`pid`),
+  CONSTRAINT `fk_1` FOREIGN KEY (`pid`) REFERENCES `test`.`parent` (`id`) ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+```
+
+## Foreign key constraint check
+
+Whether to enable the foreign key constraint check is controlled by the system variable [`foreign_key_checks`](/system-variables.md#foreign_key_checks). By default, this variable is set to `ON`, meaning that the foreign key constraint check is enabled. This variable has two scopes: `GLOBAL` and `SESSION`. Keeping this variable enabled can ensure the integrity of foreign key reference relationships.
+
+The effect of disabling [`foreign_key_checks`](/system-variables.md#foreign_key_checks) is as follows:
+
+- When you delete a parent table referenced by a foreign key, the deletion can succeed only when the foreign key constraint check is disabled.
+- When you import data to a database, the order of creating tables might be different from the foreign key dependency order, which might cause the creation of tables to fail. Only when the foreign key constraint check is disabled can the tables be created successfully. In addition, disabling the foreign key constraint check can speed up data import.
+- When you import data to a database, if the data of the child table is imported first, an error will be reported. Only when the foreign key constraint check is disabled can the data of the child table be imported successfully.
+- If an `ALTER TABLE` operation to be executed involves a change of the foreign key, this operation succeeds only when the foreign key constraint check is disabled.
+
+When the foreign key constraint check is disabled, the foreign key constraint check and reference operation are not executed, except for the following scenarios:
+
+- If the execution of `ALTER TABLE` might result in wrong definition of the foreign key, an error is still reported during the execution.
+- When deleting the index required by the foreign key, you should delete the foreign key first. Otherwise, an error is reported.
+- When you create a foreign key but it does not meet related conditions or restrictions, an error is reported.
+
+## Locking
+
+When `INSERT` or `UPDATE` a child table, the foreign key constraint checks whether the corresponding foreign key value exists in the parent table, and locks the row in the parent table to avoid the foreign key value being deleted by other operations violating the foreign key constraint. The locking behavior is equivalent to performing a `SELECT FOR UPDATE` operation on the row where the foreign key value is located in the parent table.
+
+Because TiDB currently does not support `LOCK IN SHARE MODE`, if there are a lot of concurrent writes to the child table and most of the referenced foreign key values are the same, there might be serious locking conflicts. It is recommended to disable [`foreign_key_checks`](/system-variables.md#foreign_key_checks) when writing a large number of child table data.
```

Comment:
```suggestion
Because TiDB currently does not support `LOCK IN SHARE MODE`, if a child table accepts a large number of concurrent writes and most of the referenced foreign key values are the same, there might be serious locking conflicts. It is recommended to disable [`foreign_key_checks`](/system-variables.md#foreign_key_checks) when writing a large number of child table data.
```